.github/workflows/php-test.yml

```diff
@@ -70,7 +72,39 @@ jobs:
       - uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ matrix.php }}
- 
+
+      - name: Install PHP dependencies
+        uses: ramsey/composer-install@v2
+        with:
+          composer-options: '--prefer-dist --no-progress --no-interaction --no-scripts'
+
+      # Installs a different PHPUnit version depending on the PHP version we're testing against.
+      #
+      # | WP  / PHP | PHPUnit |
+      # |-----------|---------|
+      # | *   / 7.0 | 6       |
+      # | *   / 7.1 | 7       |
+      # | *   / 7.2 | 8       |
+      # | *   / 7.3 | 9       |
+      # | *   / 7.4 | 9       |
+      # | *   / 8   | 9       |
+      #
+      # See https://make.wordpress.org/core/handbook/references/phpunit-compatibility-and-wordpress-versions/
+      - name: Update PHPUnit
+        run: |
+          if [[ $PHP_VERSION == "7.1"  ]]; then
+            echo "Installing PHPUnit 7.5x"
+            composer require --ignore-platform-reqs --no-interaction --no-scripts phpunit/phpunit:^7.5 --with-dependencies
+          elif [[ $PHP_VERSION == "7.2"  ]]; then
+            echo "Installing PHPUnit 8.5x"
+            composer require --ignore-platform-reqs --no-interaction --no-scripts phpunit/phpunit:^8.5 --with-dependencies
+          elif [[ $PHP_VERSION != "7.0"  ]]; then
+            echo "Installing latest version of PHPUnit"
+            composer update --ignore-platform-reqs --no-interaction --no-scripts yoast/phpunit-polyfills --with-dependencies
+          fi
+        env:
+          PHP_VERSION: ${{ matrix.php }}
```

Comment:
Can't we solve this in a simpler way via only `composer.json` (without hard-coding a "PHP version to PHPUnit version" map)?

Reply:
I wouldn’t know how. Composer always uses a lower version because of the PHP requirements & the lock file. This map is how I‘ve successfully done it in other projects as well. If you know a better way feel free to make edits!

Reply:
We could consider updating the `pretest-php` and `pretest-php-multisite` scripts in `package.json` to `composer update` instead of `composer install`. We are only using dependencies for workflow related things, so they don't affect the production code tested, so IMO this is fine.

I have used that approach on other WordPress plugins that needed to work in a wider PHP version range, and it works without any issues, and without manual workaround like this.

Reply:
That doesn't work. `composer update` will still just install PHPUnit 6.5 because that's the last version supporting PHP 7.0. And if you use `--ignore-platform-reqs` then it will always install PHPunit 9, which won't work on older PHP versions.

Reply:
If I'm missing something and you have precedent code, please feel free to make edits or submit a PR to this one.

Reply:
> That doesn't work. `composer update` will still just install PHPUnit 6.5 because that's the last version supporting PHP 7.0.

What is the problem with that? Maybe I'm not understanding why this is needed in the first place. If a CI workflow uses PHP 7.0, why would PHPUnit 6.5 not be correct?

Reply:
Because PHPUnit 6.5 does not run on PHP 8.

When I run composer update on my machine with PHP 8.2, I still get 6.5. I can‘t run any tests. _That‘s_ the problem.

Reply:
Why are tests passing then without this? 🤔 
In `trunk` there are no errors as far as I can tell, and it already tests with PHP 8.x as well.

Composer generally handles this kind of stuff. If PHPUnit is for some reason stuck at 6.5.x, then there's probably another dependency "holding us back" by depending on this older version.

Reply:
Oh wait, now it's clear what the problem is, the `platform` requirement in `composer.json`. Sorry, I got completely confused somehow before now, and was looking at the `performance-lab` `composer.json` 🤦 

Indeed, here we should keep the `platform` requirement in place, due to having a couple of _non-dev_ dependencies as well.

I guess the approach of force installing a specific PHPUnit version is an okay workaround, but a better solution may be to use a development specific `composer.json` without the `platform` limitation, kind of similar to how we already do in the `build-cs` folder for PHP linting.

Reply:
Well we could move all dev dependencies like phpunit to build-cs/

Reply:
In that case we should remove the `platform` from `build-cs/composer.json`, I don't think there's a good reason to have it there. Forcing a `platform` in `composer.json` is generally very wrong, the only reason we have to do it here is to ensure compatibility of the production dependencies with PHP 7.0. But for `build-cs/composer.json` that is irrelevant, we should rely on the actual PHP version used.

So I think moving all `require-dev` to that file (and removing the `platform` requirement there) is worth a try.

Reply:
And then probably also the lock file there

Reply:
OK that's even more of a mess now...

Reply:
@felixarntz so moving PHPUnit to `build-cs` is even more of a mess because there we have things like PHPStan which only support 7.2+, so we can't do `composer update` there because of that.

Reply:
At this point I would really just recommend going with what I had working yesterday and then a better option can be explored separately.